SpiffWorkflow/exceptions.py

```diff
@@ -15,23 +15,43 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 # 02110-1301  USA
+import re
 
+from SpiffWorkflow.util import levenshtein
 
-class WorkflowException(Exception):
+
+class SpiffWorkflowException(Exception):
+    """
+    Base class for all SpiffWorkflow-generated exceptions.
+    """
+    def __init__(self, msg):
+        super().__init__(msg)
+        self.notes = []
+
+    def add_note(self, note):
+        """add_note is a python 3.11 feature, this can be removed when we
+        stop supporting versions prior to 3.11"""
+        self.notes.append(note)
+
+    def __str__(self):
+        return super().__str__() + ". " + ". ".join(self.notes)
+
+
+class WorkflowException(SpiffWorkflowException):
     """
     Base class for all SpiffWorkflow-generated exceptions.
     """
 
-    def __init__(self, sender, error):
+    def __init__(self, message, sender=None):
```

Comment:
Sugestion: let's also change `sender` to `task_spec` to make it clearer what it actually is.